### PR TITLE
Add linters and do a general cleanup

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -21,6 +21,63 @@ repos:
...
+
+  # Lint: Dockerfile
+  - repo: https://github.com/hadolint/hadolint.git
+    rev: v2.12.1-beta
+    hooks:
+      - id: hadolint-docker
+        entry: hadolint/hadolint:v2.12.1-beta hadolint
+
+  # Lint: YAML
+  - repo: https://github.com/adrienverge/yamllint.git
```
Comment (manics) [inline .pre-commit-config.yaml:44]: What are the differences between what yamllint reports and the autoformatting done by prettier?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -21,6 +21,63 @@ repos:
...
+        args: ["-x"]
+
+  # Lint: Python
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+      - id: flake8
+
+  # Lint: Markdown
+  - repo: https://github.com/igorshubovych/markdownlint-cli
```
Comment (manics) [inline .pre-commit-config.yaml:72]: As with yamllint how do the reports from this compare with the markdown autoformatted by prettier?

Comment (mathbunnyru) [reply .pre-commit-config.yaml:72]: In my experience, prettier skips some errors.

For example:
```
# Heading 1

### Heading 3

We skipped out a 2nd level heading in this document
```

Or another one:
```
# This is a heading.
```

In my experience, I haven't seen these tools to conflict, and I like to use them both.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mathbunnyru) [reply .pre-commit-config.yaml:44]: To be honest, I can't find examples, when yamllint shows some errors, when prettier doesn't.
I think you might be right and this is redundant here.

Comment (mathbunnyru) [reply .pre-commit-config.yaml]: Remove yamllint.

**Comment by mathbunnyru** [comment]
@manics could you please take a look?

**Comment by mathbunnyru** [comment]
@manics done.

**Review by manics — APPROVED** [review]
Thanks!

**Pull request merged**